.github/workflows/pr_and_main.yml

```diff
@@ -16,15 +16,17 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
+      # Run this before any other commands so that git diff does not pick up formatting changes.
+      - run: |
+          find . -type f -not -path '.git/*' -exec sed -ri "s/[0-9]{4} (Scipp contributors)/$(date +%Y) \1/g" {} +
+          git diff --exit-code
       - run: sudo apt-get install --yes clang-format-10
       - run: find lib -type f -regex '.*\.\(cpp\|h\|tcc\)' -exec clang-format-10 -i {} +
-      - run: pip install cmake_format==0.6.9 flake8 yapf==0.30.0
+      - run: pip install cmake_format==0.6.9 flake8 nb-clean==2.1.0 yapf==0.30.0
       - run: find . -type f -name CMakeLists.txt -or -name '*.cmake' -not -path "./lib/.tbb/*" | xargs cmake-format -i
+      - run: find docs -type f -regex '.*\.ipynb' | xargs nb-clean clean --remove-empty-cells --preserve-cell-metadata
```

Comment:
Any reason against running this on the top level folder? There is also https://github.com/scipp/scipp/tree/main/tools/manual_testing that contains a notebook, and there may be others, e.g., if we add code documentation alongside code?

Reply:
No. I just didn't think about notebooks in other locations.